src/category_theory/limits/shapes/images.lean

```diff
@@ -273,3 +274,111 @@ end
 end
 
 end category_theory.limits
+
+namespace category_theory.limits
+
+variables {C : Type u} [𝒞 : category.{v} C]
+include 𝒞
+
+section
+
+instance {X Y : C} (f : X ⟶ Y) [has_image f] : has_image (arrow.mk f).hom :=
+by { rw arrow.mk_hom, apply_instance }
+
+end
+
+section has_image_map
+variables {f g : arrow C} [has_image f.hom] [has_image g.hom] (sq : f ⟶ g)
+
+/-- An image map is a morphism `image f → image g` fitting into a commutative square and satisfying
+    the obvious commutativity conditions. -/
+class has_image_map :=
```

Comment:
How about at the point we define `has_image_map`, we write in the arguments explicitly. For important definitions I find it confusing to have to go searching for the list of local variables before I can understand what it depends on.

Reply:
(Not explicitly as in `( ... )` as opposed to `{ ... }`, just explicitly as in not using using `variables`.)

Reply:
Good idea. Fixed.